spec/models/property_spec.rb

```diff
@@ -0,0 +1,177 @@
+      expect(property.errors[:name]).to eq ["can't be blank", "must be an ownable property in Skyrim, or the Arch-Mage's Quarters"]
+    end
+
+    it 'must have a valid hold' do
+      property.validate
+      expect(property.errors[:hold]).to eq ["can't be blank", 'must be one of the nine Skyrim holds, or Solstheim']
+    end
+
+    it 'only allows up to 10 per game', :aggregate_failures do
+      CanonicalProperty.all.each do |canonical_property|
+        game.properties.create!(
+          canonical_property: canonical_property,
+          name:               canonical_property.name,
+          hold:               canonical_property.hold,
+          city:               canonical_property.city,
+        )
+      end
+
+      property.game = game
+      property.name = 'Vlindrel Hall'
+      property.hold = 'The Reach'
+      property.validate
+      expect(property.errors[:game]).to eq ['already has max number of ownable properties']
+      expect(Rails.logger).to have_received(:error).with('Cannot create property "Vlindrel Hall" in hold "The Reach": this game already has 10 properties')
+    end
+
+    describe 'uniqueness' do
+      let(:canonical_property) { CanonicalProperty.first }
+
+      before do
+        game.properties.create!(
+          canonical_property: canonical_property,
+          name:               canonical_property.name,
+          hold:               canonical_property.hold,
+          city:               canonical_property.city,
+        )
+      end
+
+      it 'has a unique combination of game and canonical property' do
+        property.game               = game
+        property.canonical_property = canonical_property
+        property.validate
+        expect(property.errors[:canonical_property]).to eq ['must be unique per game']
+      end
+
+      it 'has a unique name per game' do
+        property.game = game
+        property.name = canonical_property.name
+        property.validate
+        expect(property.errors[:name]).to eq ['must be unique per game']
+      end
+
+      it 'has a unique hold per game' do
+        property.game = game
+        property.hold = canonical_property.hold
+        property.validate
+        expect(property.errors[:hold]).to eq ['must be unique per game']
+      end
+    end
+
+    describe 'consistency with canonical property' do
+      let(:canonical_property) { CanonicalProperty.find_by(name: 'Severin Manor') }
+
+      it 'must have the same name, hold, and city as the canonical property' do
+        property.canonical_property_id = canonical_property.id
+        property.name                  = canonical_property.name
+        property.hold                  = canonical_property.hold
+        property.city                  = nil
+        property.validate
+        expect(property.errors[:base]).to eq ['property attributes must match attributes of a property that exists in Skyrim']
+      end
+    end
+
+    describe 'arcane enchanter availability' do
```

Comment:
There really should be an 'alchemy lab availability' example group, but there are no properties in Skyrim where an alchemy lab is not a possible amenity, so it's really not that relevant anyway.